examples/README.md

```diff
@@ -267,3 +279,89 @@ Add the test runner to the scripts in your package.json
 ```
 
 Run your tests with `yarn test`
+
+# Using code examples in the docs site
+
+Creating and using code snippets is relatively straight forward. You can pull in a whole file or by adding some comments you can pull in just a particular few lines of a file as necessary.
+
+To streamline the PR process, you should create a branch in the `docs` repo to add your code snippets before your branch for the code example in `ts-immutable-sdk` is merged to main. This way we can point the docs site to your branch for the code snippets and you can see your code example in situ and make any changes required to your code example before going through the PR process.
+
+## Update import code branch
+
+In your `docs` branch modify the file `/src/remark/import-code.mjs` and update the `BRANCH` constant from `main` to the name of your branch on `ts-immutable-sdk` e.g.
+
+```
+const BRANCH = 'DVR-193-passport-signing-example';
+```
+
+Now your `docs` branch will be pulling the code examples from your branch in `ts-immutable-sdk` and you can use them locally in your `docs` branch to make sure they make sense in the page.
+
+⚠️ ⚠️ ⚠️ **IMPORTANT** ⚠️ ⚠️ ⚠️
+
+Make sure to change the branch back to `main` before merging your `docs` branch or it will **BREAK** the docs site.
+
+## Creating code snippets
+
+In your code examples in `ts-immutable-sdk` find the parts of the code you want to use as code snippets and wrap them in the `#doc` and `#enddoc` comments while providing a label.
+
+Labels have to be unique within a file so you should use a simple naming convention to avoid clashes within the file e.g.
+
+e.g. `<library>-<tag>`
+
+```
```

Comment:
Can we add syntax highlighting to all the codeblocks? For eg here and in similar places we could do ```ts/```js, and in other parts we could also add the ```bash syntax highlighting.